### PR TITLE
[KOGITO-8222] Test parallel examples build

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-8222

Related:
- https://github.com/kiegroup/kogito-pipelines/pull/690
- https://github.com/kiegroup/drools/pull/4808
- https://github.com/kiegroup/kogito-runtimes/pull/2606
- https://github.com/kiegroup/kogito-apps/pull/1502
- https://github.com/kiegroup/kogito-examples/pull/1410

### CODE REVIEW

File: .github/workflows/kogito-examples-pr.yml [hunk 1]
```diff
@@ -23,9 +23,10 @@ jobs:
         os: [ubuntu-latest]
         java-version: [11]
         maven-version: ['3.8.6']
+        examples-subfolder: [ serverless-workflow-examples, kogito-quarkus-examples ]
```
Comment (cristianonicolai) [inline .github/workflows/kogito-examples-pr.yml]: \+ springboot

Comment (ricardozanini) [reply .github/workflows/kogito-examples-pr.yml]: I think this makes sense, since sb can be widely used by dmn/process/rules

Comment (radtriste) [reply .github/workflows/kogito-examples-pr.yml]: yeah yeah, that was planned :D It was just a test for now :P
I would like to make it more dynamic and scan the first level directories so we don't have each time to maintain the directories to tests. wdyt?

Comment (radtriste) [reply .github/workflows/kogito-examples-pr.yml]: for now will be static, dynamic may come in a future PR

**Comment by radtriste** [comment]
jenkins test this

**Comment by radtriste** [comment]
jenkins run native

**Comment by radtriste** [comment]
jenkins retest this

**Comment by radtriste** [comment]
jenkins retest this

**Comment by radtriste** [comment]
jenkins run native

**Comment by radtriste** [comment]
old build times (full kogito-examples):
- tests -> 1h40
- native -> 3h25

new times max:
- 50m (kogito-quarkus-examples)
- 2h05 (kogito-quarkus-examples)

**Review by ricardozanini — APPROVED** [review]

**Review by Ginxo — APPROVED** [review]
LGTM
I'm just worried about `Concurrent Jobs` limitation [1] for the plan we have

[1] https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits

**Pull request merged**